light/provider/provider.go

```diff
@@ -9,8 +9,9 @@ import (
 // Provider provides information for the light client to sync (verification
 // happens in the client).
 type Provider interface {
-	// ChainID returns the blockchain ID.
-	ChainID() string
+	// String returns an identifier for the provider. This should be unique.
+	// This is only used for logging and returning errors.
+	String() string
```

Comment:
this probably should not be required by the interface

Reply:
if it's only used for logging then it's fine to not require it from implementations and simply do `fmt.Sprintf("%v", provider)` which will call `String()` if such is defined by the struct.

Reply:
ok yeah I was unsure whether we should enforce that the provider also adhere to the stringer interface. I will remove it